---
cve: 2014-0472
title: "Django: unexpected code execution using reverse()"
description: >
    A flaw was found in the way Django's reverse() URL resolver function
    constructed certain URLs. A remote attacker able to request a specially
    crafted view from a Django application could use this flaw to import
    and execute arbitrary Python modules on the system under the privileges
    of the user running the application.
cvss_v2: 4.3
references:
    - https://www.djangoproject.com/weblog/2014/apr/21/security/
    - https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2014-0472
    - https://rhn.redhat.com/errata/RHSA-2014-0456.html
    - https://www.debian.org/security/2014/dsa-2934
affected:
    - name: Django
      version:
        - "<=1.7b1"
        - "<=1.6.2"
        - "<=1.5.5"
        - "<=1.4.10"
      fixedin:
        - ">=1.7b2"
        - ">=1.6.3"
        - ">=1.5.6"
        - ">=1.4.11"